### PR TITLE
Full upgrade

### PR DESCRIPTION
**Pull request by yalh76**
## Problem
- *Full upgrade*

## Solution
- [x] *Apply example_ynh*
- [x] *Implement ynh_add_secure_repos__3*
- [x] *Add service in the admin panel*
- [x] *Implement ynh_systemd_action*
- [x] *Cleanup scripts*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
- [ ] **Code review**
- [ ] **Approval (LGTM)**  
*Code review and approval have to be from a member of @YunoHost/apps group*
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/peertube_ynh_testing%20(yalh76)/)

When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,17 +17,18 @@ But it's not enough because one video could become famous and overload the serve
...
 Thanks to [WebTorrent](https://github.com/feross/webtorrent), we can make P2P (thus BitTorrent) inside the web browser, as of today.
 
-## Why is that cool ?
+### Why is that cool ?
 Servers are run independently by different people and organizations. They can apply wildly different moderation policies, so you can find or make one that fits your taste perfectly.
 
 By watching a video, you help the hosting provider to broadcast it by becoming a broadcaster of the video yourself. Each instance doesn’t need much money to broadcast the videos of its users.
 
-## Demonstration
+Shipped Version: **1.2.1**
```
Comment (anmol26s) [inline README.md]: Is not good to have shipped version on top?

**Comment by yalh76** [comment]
> 
> 
> Should we use ynh_add_secure_repos before it gets into stable? I ask because I think it need more testing till it should be use in the apps. It will need to change again and again as it gets updated.

As you want but ynh_add_secure_repos__3 do a clean job and work well, already tested and used on 3 other packages.

File: README.md (hunk 1 above)
Comment (yalh76) [reply README.md]: I can put the shipped version on top, but I though that more important than the shipped version, was the text about what is Peertube.

Let me know where you want it, and I put it.

**Comment by anmol26s** [comment]
Ok then we can go with it. Is it ready for manual testing?
There is other branch for people who get error on yarn while installing/upgrading. Should these changes be pushed into that branch too? There maybe some people using this branch( at least 2 I knew, but not sure if they use it).

https://github.com/YunoHost-Apps/peertube_ynh/tree/ovh_fix

File: README.md (hunk 1 above)
Comment (anmol26s) [reply README.md]: I don't have any problem with it, as it don't hinder the working of the package.

**Comment by yalh76** [comment]
> 
> 
> Ok then we can go with it. Is it ready for manual testing?
> There is other branch for people who get error on yarn while installing/upgrading. Should these changes be pushed into that branch too? There maybe some people using this branch( at least 2 I knew, but not sure if they use it).
> 
> https://github.com/YunoHost-Apps/peertube_ynh/tree/ovh_fix

For the other branches, I don't know, I'ven't checked them.
Maybe, when this PR will be merged in testing, do a PR from testing to the branche ovh_fix and ask them to test, if it's OK, then you will be able to merge ovh_fix to testing to add the changes made

**Comment by yalh76** [comment]
> 
> 
> Ok then we can go with it. Is it ready for manual testing?

Yes ready for testing :)

File: README.md (hunk 1 above)
Comment (yalh76) [reply README.md]: Well, Neither do I, so if you want I moved the shipped version on top  or if you are OK to let it here, I let you resolve the conversation ;)

**Comment by anmol26s** [comment]
Why not apply this
`path_url=$(ynh_app_setting_get $app path) ` to upgrade script https://github.com/YunoHost-Apps/peertube_ynh/pull/85/commits/28c979994d7db4abeb3b3d112d09a4021ab034c8#diff-44cb16c778719320333118c04d509a7cR21 when this is applied to restore script https://github.com/yalh76/peertube_ynh/blob/testing/scripts/restore#L32  ?

**Comment by anmol26s** [comment]
This https://github.com/yalh76/peertube_ynh/blob/testing/scripts/upgrade#L98 should be like this 
https://github.com/YunoHost-Apps/peertube_ynh/blob/master/scripts/restore#L76-L79 too.

**Comment by anmol26s** [comment]
https://github.com/YunoHost-Apps/peertube_ynh/blob/master/check_process#L30 should be `na`

**Comment by yalh76** [comment]
> 
> 
> Why not apply this
> `path_url=$(ynh_app_setting_get $app path) ` to upgrade script [28c9799#diff-44cb16c778719320333118c04d509a7cR21](https://github.com/YunoHost-Apps/peertube_ynh/commit/28c979994d7db4abeb3b3d112d09a4021ab034c8#diff-44cb16c778719320333118c04d509a7cR21) when this is applied to restore script https://github.com/yalh76/peertube_ynh/blob/testing/scripts/restore#L32 ?

Correction made

**Comment by yalh76** [comment]
> e like this
> https://github.com/YunoHost-Apps/peertube_ynh/blob/master/scripts/restore#L76-L79 too.

I made no changes on that part of the code, I haven't even read that part of code :///

**Comment by yalh76** [comment]
> 
> 
> https://github.com/YunoHost-Apps/peertube_ynh/blob/master/check_process#L30 should be `na`

There are some changes on package_check and levels.
Level 4 is no more used for LDAP and SSO. 
It's why I made no changes on the levels

**Comment by anmol26s** [comment]
> I made no changes on that part of the code, I haven't even read that part of code :///

This looks my mistake, as this is done me before your PR. I did not look all your commits, but the final commit that is why got confused.

**Comment by anmol26s** [comment]
Only one thing more. It would be really nice if we could test the upgrade from previous commit, so that the upgrade don't fail. You can do it manually if you want.

upgrade=1	from_commit=cc324108b47f3f9f862f03a82eac1370ea11e246

**Comment by yalh76** [comment]
> 
> 
> Only one thing more. It would be really nice if we could test the upgrade from previous commit, so that the upgrade don't fail. You can do it manually if you want.
> 
> upgrade=1 from_commit=cc324108b47f3f9f862f03a82eac1370ea11e246

Change made, waiting package_check results

**Comment by anmol26s** [comment]
Merging this.
Thanks to @yalh76 for his work and contribution. :heart:

**Pull request merged**